### ISSUE TITLE
Javadoc summary not processed inside html tags

### ISSUE THREAD

**Issue by AkMo3** [checkstyle/checkstyle#9806]
I have read check documentation: https://checkstyle.org/config_javadoc.html#SummaryJavadoc
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
$ javac Test1.java 

$ cat Test1.java 
package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;

/**
 * Config: default.
 */
public class Test1 {

    /**
     * <p><b>Summary Normal Javadoc.</b></p>
     */
    public void foo1() {} // ok
}

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="SummaryJavadocCheck"/>
  </module>
</module>

$ javadoc Test1.java 
Loading source file Test1.java...
Constructing Javadoc information...
Standard Doclet version 14.0.2
Building tree for all the packages and classes...
Generating ./com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/Test1.html...
Generating ./com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/package-summary.html...
Generating ./com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/package-tree.html...
Generating ./constant-values.html...
Building index for all the packages and classes...
Generating ./overview-tree.html...
Generating ./deprecated-list.html...
Generating ./index-all.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./system-properties.html...
Generating ./index.html...
Generating ./help-doc.html...

$ java -jar checkstyle-8.41.1-all.jar -c config.xml Test1.java 
Starting audit...
[ERROR] /home/akmo/GitHub/Test/InlineJavadoc/Test1.java:8: Summary javadoc is missing. [SummaryJavadoc]
Audit done.
Checkstyle ends with 1 errors.

```
---

**Describe what you expect in detail.**

The summary should not raise any violation

```
$ java -jar checkstyle-8.41.1-all.jar -c config.xml Test1.java
Starting audit...
Audit done.
```
![Screenshot from 2021-04-03 00-41-19](https://user-images.githubusercontent.com/69365878/113446453-70a4ba80-9415-11eb-9959-e14774fc8330.png)

---

**Comment by rajatshrm648** [comment]
Hello @AkMo3  I want to help you in this issue

**Comment by AkMo3** [comment]
@rajatshrm648 Thanks for your interest to contribute to the project. I would advise you first to solve some easy or gsoc first and second issues before working on checks so that you get enough experience about the project. Secondly, this issue is still not approved so, we should not work for a fix for this and rather focus on already pending approved issues. 

Welcome to the community

**Comment by rajatshrm648** [comment]
Thank you for your response
I want to be a part of this project and put all my efforts to contribute to
the project

**Comment by angelsjiang** [comment]
@pbludov @AkMo3 
Hello, I was looking for some issues I could work on with my team these days, and I was looking at this issue these couple days and tried different things. I used the example provided in this post to run my test, and I took a look at the SummaryJavadocCheck documentation, the test suite, and the test resources that are related to this issue. After all I think the example Test1 that the post provided is just missing a @summary tag around the html tags, so it's not really a bug with the system.

I tried with this input file:
```
package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;

/**
 * Config: default.
 */
public class Test1 {

    /**
     * {@summary <p><b>Summary Normal Javadoc.</b></p>}
     */
    public void foo1() {} // ok
}
```
and this:
```
package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;

/**
 * Config: default.
 */
public class Test1 {

    /**
     *  <p><b>{@summary Summary Normal Javadoc.}</b></p>
     */
    public void foo1() {} // ok
}
```
And it doesn't produce the error anymore. And according to the documentation, it looks like you do need to provide the @summary tag.
But given the fact that I really just started looking into this open source project these few weeks ago, I might be missing something, so I'm not 100% sure. But just want to let you know.

**Comment by AkMo3** [comment]
@angelsjiang Firstly, `@summary` tag was introduced in JDK 10, so projects who are running a lower JDK will not be able to do this change. Secondly, it is not appropriate that users have to make amendments in their code which can be huge in projects for something that we do not support just because there is a workaround. We should be the ones doing code change in most cases, not them. We must try to support as many possibilities as possible.

**Comment by sneharani1994** [comment]
Hi @AkMo3 ,
 I’d like to work on this issue. I’ll try to reproduce and propose a fix. Please let me know if anyone is currently working on it.

**Comment by romani** [comment]
Please [read](https://github.com/checkstyle/checkstyle/blob/master/.github/CONTRIBUTING.md#getting-started) and watch videos at [Starting_Development](https://checkstyle.org/beginning_development.html#Starting_Development).
it should help to start quicker.

**Comment by sneharani1994** [comment]
Thank You,
 I’ll go through the Starting Development guide and set up my environment. I’ll update once I’m able to reproduce the issue locally.

**Comment by vivek-0509** [comment]
@sneharsh1994 Are you still working on this ??

**Comment by romani** [comment]
If no PR for a week, https://github.com/checkstyle/checkstyle/issues?q=is%3Apr+is%3Aopen+author%3Asneharsh1994 , comment "I am on it" can be ignored

**Comment by vivek-0509** [comment]
Sure thanks, i am on it

**Comment by Carbon14-48** [comment]
Hi @vivek-0509 , are you currently working on this issue? If not, I'd like to pick it up. Thank you!

**Comment by vivek-0509** [comment]
@Carbon14-48 yes, i need some clarification on it from the maintainer but since there are a lot of prs coming so it is getting bit delayed so I just converted it into draft, please choose some other issues
